### DISCUSSION TITLE
Why function jaccard_coefficient didnt return the coefficient of each pair?

### DISCUSSION THREAD

**Discussion by kukukumiko-git** [networkx/networkx#5998] [Q&A] [answered]
Hello, I'm  learning networkx now, and use a simple network as a sample

![graph](https://user-images.githubusercontent.com/114204641/191886503-5f60d737-8671-4d0f-92c0-142b5795c3d4.png)

And try to calculate the jaccard coefficient by useing function jaccard_coefficient

```
jaccard=list(nx.jaccard_coefficient(g))
print(jaccard)
```
And it return me:
`[('6', '2', 0.2), ('6', '8', 1.0), ('6', '4', 0.2), ('6', '5', 0.16666666666666666), ('6', '3', 0.0), ('2', '9', 0.16666666666666666), ('2', '8', 0.2), ('2', '4', 1.0), ('2', '7', 0.0), ('8', '4', 0.2), ('8', '3', 0.0), ('8', '5', 0.16666666666666666), ('9', '4', 0.16666666666666666), ('9', '3', 0.0), ('9', '5', 0.14285714285714285), ('4', '7', 0.0), ('5', '7', 0.0), ('3', '7', 0.0), ('3', '1', 0.5), ('1', '7', 0.5)]`
as same as this picture ( just ignore the "['1']")
![捕获](https://user-images.githubusercontent.com/114204641/191887184-2066a02e-8f9d-44b2-86d9-7a5030a5aa81.JPG)

Why it didnt return the coefficient of each pair?
I checked it by printing neighbor nodes of each node, ensure the graph was correctly read.
Thank you

**Comment by boothby** [comment, marked as answer]
Please see the documentation of the function:

https://networkx.org/documentation/stable/reference/algorithms/generated/networkx.algorithms.link_prediction.jaccard_coefficient.html

> ebunch : iterable of node pairs, optional (default = None)
>
>     Jaccard coefficient will be computed for each pair of nodes given in the iterable. 
>     The pairs must be given as 2-tuples (u, v) where u and v are nodes in the graph. 
>     If ebunch is None then all non-existent edges in the graph will be used.
>     Default value: None.

**Reply by kukukumiko-git** [reply]
Thank you

I had read the documentation ,and tried to
`    for i in g.nodes:
for j in g.nodes:
test = list(nx.jaccard_coefficient(g,(i,j)))
print(test)`
but 
`ValueError: not enough values to unpack (expected 2, got 1)`

I didnt know what's wrong, and thought use function incorrectly.......
Forgive me I'm also rookie for Python

**Reply by boothby** [reply]
The `ebunch` parameter needs to be an iterable of 2-tuples.  Try

```python
for i in g.nodes:
    for j in g.nodes:
        ebunch = [(i, j)]
        test = list(nx.jaccard_coefficient(g, ebunch))
        print(test)
```

**Reply by kukukumiko-git** [reply]
Thank you
